File: frontend/campus-erp-ui/src/app/features/academic-settings/models/update-academic-configuration-request.ts
export interface UpdateAcademicConfigurationRequest {
  // Academic Structure

  academicTermType: number;

  academicTermsPerSession: number;

  autoPromoteEnabled: boolean;

  // Attendance Rules

  minimumAttendancePercentage: number;

  allowAttendanceEditing: boolean;

  attendanceEditWindowDays: number;

  // Attendance Automation

  autoGenerateAttendanceSessions: boolean;

  autoGenerateAttendanceRecords: boolean;

  // Attendance Lock

  attendanceLockAfterDays: number;

  allowTeacherAttendanceUnlock: boolean;

  // Attendance Behaviour

  lateThresholdMinutes: number;

  medicalLeaveCountsAsPresent: boolean;

  onDutyCountsAsPresent: boolean;

  // Student Requests

  allowStudentAttendanceCorrection: boolean;
}

File: backend/CampusERP.Contracts/Requests/UpdateAcademicConfigurationRequest.cs
using CampusERP.Shared.Enums;
using System.ComponentModel.DataAnnotations;

namespace CampusERP.Contracts.Requests;

public class UpdateAcademicConfigurationRequest
{
    #region Academic Structure

    [Required]
    public AcademicTermType AcademicTermType { get; set; }

    [Range(1, 12)]
    public int AcademicTermsPerSession { get; set; }

    public bool AutoPromoteEnabled { get; set; }

    #endregion

    #region Attendance Rules

    [Range(0, 100)]
    public int MinimumAttendancePercentage { get; set; }

    public bool AllowAttendanceEditing { get; set; }

    [Range(0, 365)]
    public int AttendanceEditWindowDays { get; set; }

    #endregion

    #region Attendance Automation

    public bool AutoGenerateAttendanceSessions { get; set; }

    public bool AutoGenerateAttendanceRecords { get; set; }

    #endregion

    #region Attendance Lock

    [Range(0, 365)]
    public int AttendanceLockAfterDays { get; set; }

    public bool AllowTeacherAttendanceUnlock { get; set; }

    #endregion

    #region Attendance Behaviour

    [Range(0, 180)]
    public int LateThresholdMinutes { get; set; }

    public bool MedicalLeaveCountsAsPresent { get; set; }

    public bool OnDutyCountsAsPresent { get; set; }

    #endregion

    #region Student Requests

    public bool AllowStudentAttendanceCorrection { get; set; }

    #endregion
}